dex/tests/close-markets/programs/close-markets/src/lib.rs

```diff
@@ -0,0 +1,338 @@
+        bump = bumps.usdc_mint,
+        payer = payer
+    )]
+    pub usdc_mint: Box<Account<'info, Mint>>,
+    #[account(
+        init,
+        mint::decimals = 6,
+        mint::authority = payer,
+        seeds = [b"serum_mint".as_ref()],
+        bump = bumps.serum_mint,
+        payer = payer
+    )]
+    pub serum_mint: Box<Account<'info, Mint>>,
+    #[account(
+        init,
+        seeds = [SERUM_MARKET_SEED.as_ref()],
+        bump = bumps.serum_market,
+        space = SERUM_MARKET_SPACE, 
+        payer = payer,
+        owner = dex_program.key(
+    ))]
+    pub serum_market: UncheckedAccount<'info>,
+    #[account(
+        init,
+        seeds = [REQUEST_QUEUE_SEED.as_ref()],
+        bump = bumps.request_queue,
+        space = REQUEST_QUEUE_SPACE,
+        payer = payer,
+        owner = dex_program.key(
+    ))]
+    pub request_queue: UncheckedAccount<'info>,
+    #[account(
+        init,
+        token::mint = serum_mint,
+        token::authority = vault_signer,
+        seeds = [COIN_VAULT_SEED.as_ref()],
+        bump = bumps.coin_vault,
+        payer = payer
+    )]
+    pub coin_vault: Box<Account<'info, TokenAccount>>,
+    #[account(
+        init,
+        token::mint = usdc_mint,
+        token::authority = vault_signer,
+        seeds = [PC_VAULT_SEED.as_ref()],
+        bump = bumps.pc_vault,
+        payer = payer
+    )]
+    pub pc_vault: Box<Account<'info, TokenAccount>>,
+    // TODO probably no way to verify these seeds since it uses a different heuristic
+    pub vault_signer: UncheckedAccount<'info>,
+    // These shouldn't need to be signers
+    #[account(mut)]
+    pub event_queue: UncheckedAccount<'info>,
+    #[account(mut)]
+    pub bids: UncheckedAccount<'info>,
+    #[account(mut)]
+    pub asks: UncheckedAccount<'info>,
+    pub system_program: Program<'info, System>,
+    pub token_program: Program<'info, Token>,
+    // TODO check this via anchor-spl?
+    pub dex_program: UncheckedAccount<'info>,
+    pub rent: Sysvar<'info, Rent>,
+}
+
+impl<'info> InitializeMarket<'info> {
+    pub fn initialize_serum_market_ctx(&self) -> CpiContext<'_, '_, '_, 'info, dex::InitializeMarket<'info>> {
+        let cpi_accounts = dex::InitializeMarket {
+            market: self.serum_market.to_account_info(),
+            req_q: self.request_queue.to_account_info(),
+            event_q: self.event_queue.to_account_info(),
+            bids: self.bids.to_account_info(),
+            asks: self.asks.to_account_info(),
+            coin_vault: self.coin_vault.to_account_info(),
+            pc_vault: self.pc_vault.to_account_info(),
+            coin_mint: self.serum_mint.to_account_info(),
+            pc_mint: self.usdc_mint.to_account_info(),
+            rent: self.rent.to_account_info(),
+        };
+        let cpi_ctx = CpiContext::new(self.dex_program.to_account_info(), cpi_accounts);
+        // prune_auth will be the open orders auth and prune auth
+        cpi_ctx.with_remaining_accounts(vec![
```

Comment:
probably needs cargo fmt

Reply:
i ran cargo format and it only caught one error, a trailing space left somewhere.